src/app/shared/components/admin-carriages/carriage-form/carriage-form.component.html

```diff
@@ -0,0 +1,72 @@
+<mat-card>
+  <mat-card-title>
+    <ng-container *ngIf="!carriage">Create Carriage</ng-container>
+    <ng-container *ngIf="carriage">Edit Carriage: {{ carriage.name }}</ng-container></mat-card-title
+  >
+  <mat-card-content>
+    <form [formGroup]="carriageForm" (ngSubmit)="onSubmit()">
+      <mat-form-field appearance="outline">
+        <mat-label>Rows Count</mat-label>
+        <input matInput type="number" formControlName="rowCount" />
+        <mat-error *ngIf="carriageForm.get('rowCount')?.hasError('required')">
```

Comment:
And I ask you to refactor error message  like make it Sasha here https://github.com/mogitrash/ng-train/tree/feat/admin-carriages/src/app/pages/signin-page or check this example with email https://material.angular.io/components/form-field/overview#form-field-error . It will be better to have this logic in component

Reply:
This form doesn't have as strict criteria as the registration form, but I'll try to do as you ask.